<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PostGISConnectorDialogForm</class>
 <widget class="QDialog" name="PostGISConnectorDialogForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>575</width>
    <height>495</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PostGIS Connector</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_10">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout_9">
     <item row="0" column="0">
      <widget class="QFrame" name="frame_2">
       <property name="styleSheet">
        <string notr="true">QWidget { background: white }</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLabel" name="m_datasourceTypeTitleLabel">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>PostGIS</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="1" column="0">
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QGroupBox" name="groupBox">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Base Information</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_12">
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_8">
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Title</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLineEdit" name="m_datasourceTitleLineEdit"/>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <layout class="QGridLayout" name="gridLayout_11">
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Description</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QTextEdit" name="m_datasourceDescriptionTextEdit">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>60</height>
                 </size>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QGroupBox" name="groupBox_2">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Connection Information</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_29">
            <property name="leftMargin">
             <number>9</number>
            </property>
            <property name="topMargin">
             <number>9</number>
            </property>
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_28">
              <item row="0" column="0">
               <layout class="QGridLayout" name="gridLayout_26">
                <item row="0" column="0">
                 <layout class="QGridLayout" name="gridLayout_19">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_8">
                    <property name="text">
                     <string>Host</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="m_hostNameLineEdit">
                    <property name="minimumSize">
                     <size>
                      <width>110</width>
                      <height>0</height>
                     </size>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="1" column="0">
                 <layout class="QGridLayout" name="gridLayout_20">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_6">
                    <property name="text">
                     <string>User</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="m_userNameLineEdit">
                    <property name="minimumSize">
                     <size>
                      <width>110</width>
                      <height>0</height>
                     </size>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="2" column="0">
                 <layout class="QGridLayout" name="gridLayout_24">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_10">
                    <property name="text">
                     <string>Database Name</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QComboBox" name="m_databaseComboBox">
                    <property name="minimumSize">
                     <size>
                      <width>110</width>
                      <height>0</height>
                     </size>
                    </property>
                    <property name="editable">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item row="0" column="1">
               <layout class="QGridLayout" name="gridLayout_27">
                <item row="0" column="0">
                 <layout class="QGridLayout" name="gridLayout_22">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>Port</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <layout class="QGridLayout" name="gridLayout_21">
                    <item row="0" column="0">
                     <widget class="QLineEdit" name="m_portLineEdit">
                      <property name="minimumSize">
                       <size>
                        <width>50</width>
                        <height>0</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>80</width>
                        <height>16777215</height>
                       </size>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="1">
                     <spacer name="horizontalSpacer_4">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>40</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
                <item row="1" column="0">
                 <layout class="QGridLayout" name="gridLayout_23">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_7">
                    <property name="text">
                     <string>Password</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="m_passwordLineEdit">
                    <property name="minimumSize">
                     <size>
                      <width>110</width>
                      <height>0</height>
                     </size>
                    </property>
                    <property name="contextMenuPolicy">
                     <enum>Qt::DefaultContextMenu</enum>
                    </property>
                    <property name="echoMode">
                     <enum>QLineEdit::Password</enum>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="2" column="0">
                 <layout class="QGridLayout" name="gridLayout_25">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_9">
                    <property name="text">
                     <string>Client Encoding</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QComboBox" name="m_clientEncodingComboBox">
                    <property name="minimumSize">
                     <size>
                      <width>110</width>
                      <height>0</height>
                     </size>
                    </property>
                    <property name="editable">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QCheckBox" name="m_savePasswordCheckBox">
              <property name="text">
               <string>Save Password</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="m_advancedConnectionOptionsCheckBox">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Advanced Connection Options</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <widget class="QGroupBox" name="m_advancedConnectionOptionsGroupBox">
         <property name="minimumSize">
          <size>
           <width>243</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>243</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Advanced Options</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_38">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QScrollArea" name="scrollArea">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="lineWidth">
             <number>1</number>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents_2">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>241</width>
               <height>365</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout_37">
              <item row="0" column="0">
               <layout class="QGridLayout" name="gridLayout_36">
                <item row="0" column="0">
                 <widget class="QGroupBox" name="groupBox_4">
                  <property name="title">
                   <string>Tables</string>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_32">
                   <item row="0" column="0">
                    <layout class="QGridLayout" name="gridLayout_31">
                     <item row="0" column="0">
                      <layout class="QGridLayout" name="gridLayout_30">
                       <item row="0" column="0">
                        <widget class="QLabel" name="label_11">
                         <property name="text">
                          <string>Tables to be hidden</string>
                         </property>
                        </widget>
                       </item>
                       <item row="1" column="0">
                        <widget class="QLineEdit" name="m_tablesToHideLineEdit"/>
                       </item>
                      </layout>
                     </item>
                     <item row="1" column="0">
                      <widget class="QCheckBox" name="m_hideMetadataTablesCheckBox">
                       <property name="text">
                        <string>Hide spatial metadata tables</string>
                       </property>
                      </widget>
                     </item>
                     <item row="2" column="0">
                      <widget class="QCheckBox" name="m_hideRasterTablesCheckBox">
                       <property name="text">
                        <string>Hide raster tables</string>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item row="1" column="0">
                 <layout class="QGridLayout" name="gridLayout_4">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_12">
                    <property name="text">
                     <string>Connect timeout</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <layout class="QGridLayout" name="gridLayout_3">
                    <item row="0" column="0">
                     <widget class="QSpinBox" name="m_connectTimeoutSpinBox">
                      <property name="minimum">
                       <number>5</number>
                      </property>
                      <property name="maximum">
                       <number>60</number>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="1">
                     <spacer name="horizontalSpacer_2">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>40</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
                <item row="2" column="0">
                 <widget class="QGroupBox" name="groupBox_5">
                  <property name="title">
                   <string>Connection Pooling</string>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_34">
                   <item row="0" column="0">
                    <layout class="QGridLayout" name="gridLayout_33">
                     <item row="0" column="0">
                      <layout class="QGridLayout" name="gridLayout_5">
                       <item row="0" column="0">
                        <widget class="QLabel" name="label_13">
                         <property name="text">
                          <string>Max</string>
                         </property>
                        </widget>
                       </item>
                       <item row="1" column="0">
                        <widget class="QSpinBox" name="m_maxPoolSizeSpinBox">
                         <property name="minimum">
                          <number>1</number>
                         </property>
                         <property name="maximum">
                          <number>1024</number>
                         </property>
                         <property name="value">
                          <number>4</number>
                         </property>
                        </widget>
                       </item>
                      </layout>
                     </item>
                     <item row="0" column="1">
                      <layout class="QGridLayout" name="gridLayout_6">
                       <item row="0" column="0">
                        <widget class="QLabel" name="label_14">
                         <property name="text">
                          <string>Min</string>
                         </property>
                        </widget>
                       </item>
                       <item row="1" column="0">
                        <widget class="QSpinBox" name="m_minPoolSizeSpinBox">
                         <property name="minimum">
                          <number>1</number>
                         </property>
                         <property name="maximum">
                          <number>1024</number>
                         </property>
                         <property name="value">
                          <number>1</number>
                         </property>
                        </widget>
                       </item>
                      </layout>
                     </item>
                     <item row="0" column="2">
                      <layout class="QGridLayout" name="gridLayout_7">
                       <item row="0" column="0">
                        <widget class="QLabel" name="label_15">
                         <property name="text">
                          <string>Max Idle Time</string>
                         </property>
                        </widget>
                       </item>
                       <item row="1" column="0">
                        <widget class="QSpinBox" name="m_maxIdleTimeSpinBox">
                         <property name="minimum">
                          <number>60</number>
                         </property>
                         <property name="maximum">
                          <number>3600</number>
                         </property>
                        </widget>
                       </item>
                      </layout>
                     </item>
                     <item row="0" column="3">
                      <spacer name="horizontalSpacer_6">
                       <property name="orientation">
                        <enum>Qt::Horizontal</enum>
                       </property>
                       <property name="sizeHint" stdset="0">
                        <size>
                         <width>40</width>
                         <height>20</height>
                        </size>
                       </property>
                      </spacer>
                     </item>
                    </layout>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item row="3" column="0">
                 <layout class="QGridLayout" name="gridLayout_35">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_16">
                    <property name="text">
                     <string>Options</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="m_optionsLineEdit"/>
                  </item>
                 </layout>
                </item>
                <item row="4" column="0">
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="Line" name="line_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1,0">
       <item>
        <widget class="te::qt::widgets::HelpPushButton" name="m_helpPushButton">
         <property name="text">
          <string>&amp;Help</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="m_testPushButton">
           <property name="text">
            <string>&amp;Test</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="m_openPushButton">
           <property name="text">
            <string>&amp;Open</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="m_closePushButton">
           <property name="text">
            <string>&amp;Close</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>te::qt::widgets::HelpPushButton</class>
   <extends>QPushButton</extends>
   <header>terralib/qt/widgets/help/HelpPushButton.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_datasourceTitleLineEdit</tabstop>
  <tabstop>m_datasourceDescriptionTextEdit</tabstop>
  <tabstop>m_hostNameLineEdit</tabstop>
  <tabstop>m_portLineEdit</tabstop>
  <tabstop>m_userNameLineEdit</tabstop>
  <tabstop>m_passwordLineEdit</tabstop>
  <tabstop>m_databaseComboBox</tabstop>
  <tabstop>m_clientEncodingComboBox</tabstop>
  <tabstop>m_advancedConnectionOptionsCheckBox</tabstop>
  <tabstop>m_openPushButton</tabstop>
  <tabstop>m_closePushButton</tabstop>
  <tabstop>m_testPushButton</tabstop>
  <tabstop>m_helpPushButton</tabstop>
  <tabstop>m_maxPoolSizeSpinBox</tabstop>
  <tabstop>m_minPoolSizeSpinBox</tabstop>
  <tabstop>m_maxIdleTimeSpinBox</tabstop>
  <tabstop>m_optionsLineEdit</tabstop>
  <tabstop>m_hideRasterTablesCheckBox</tabstop>
  <tabstop>m_tablesToHideLineEdit</tabstop>
  <tabstop>m_hideMetadataTablesCheckBox</tabstop>
  <tabstop>m_connectTimeoutSpinBox</tabstop>
  <tabstop>scrollArea</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>m_closePushButton</sender>
   <signal>pressed()</signal>
   <receiver>PostGISConnectorDialogForm</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>895</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>890</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
